### PR TITLE
Improve syntax of Json configuration file of ROBFW-AIO

### PR DESCRIPTION
**Pull request by namsonx**

### CODE REVIEW

**Comment by test-fullautomation** [comment]
Hi Son,
thank you for you pull-request.
This I will merge for coming 0.5.3 version.
Please provide also selftest and documentation update.
Thank you,
 Thomas

**Review by HolQue — COMMENTED** [review]
If the key value is of type string it makes no difference if the complete expression is encapsulated in quotes or not:
(1) "${params}['global']['string_val']"
(2) ${params}['global']['string_val']
This is OK, but the user should be aware of that - therefore this should be mentioned anywhere in the documentation, maybe as recommendation that the quotes are not necessary in this case.

Keys of type int, float and bool are recognized correctly (when expression is without quotes). With quotes the values are of type string correctly.

Keys that are None or null are also recognized correctly (but independently from quotes around the expression).

Also with quotes ("${params}['global']['null_val']") the value is of type NoneType. But in my opinion it would be more consistent when with quotes the type would be string.

**Comment by test-fullautomation** [comment]
Hi Holger,
thank you for fast testing!
Yes, this you found an issue. null and None must be string (text) if they are set in quotes.
@namsonx : Please fix this.
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

     I have just run the test with debug mode, and I observed the value of "${params}['global']['null_val']" is converted to string and remained until it is set by BuiltIn().set_global_variable("${%s}" % k.strip(), v)
     I could not attach the screenshot of my debug session to this github comment, I will show you in the sync up meeting.
     So, the root cause of the case which Holger mentioned comes from the core of ROBFW, I think!

Thank you, 
Son

**Comment by test-fullautomation** [comment]
Hi Son,
unfortunately the SelfTest and Documentation update is still missing.
By when can I expect this?
We disussed a lot on this one month ago. Is there something remaining open / or a dependency? What about the "None" issue?
Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

     I had checked the case with "None" issue which Holger mentioned, and I recognized that this issue also occurs with "True" and "False" cases.
     I had updated the commit to correct them all in this pull request.

     For the Selftest documentation of this JsonPreprocessor repo. I will update in another pull-request.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
unfortunately the SelfTest and Documentation update is still missing.
By when can I expect this?
Thank you,
Thomas

File: packagedoc/additional_docs/Description.rst [hunk 1]
```diff
@@ -12,13 +12,401 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 
-The json preprocessor:
+How to install
+--------------
 
-.. image:: ./pictures/python3-jsonpreprocessor.png
+Firstly, clone `python-jsonpreprocessor <https://github.com/test-fullautomation/python-jsonpreprocessor>`_
```
Comment (test-fullautomation) [inline packagedoc/additional_docs/Description.rst:18]: Hi Son,
pease refer to genpackagedoc. I sent some  weeks ago a mail that genpackagedoc shall be the reference.
first we propose ```pip install```.
Please make this document similar.
Thank you,
 Thomas

**Comment by test-fullautomation** [comment]
Hi Son,
thank you for the update.
Please look into my findings.
Thank you,
 Thomas

File: packagedoc/additional_docs/Description.rst (hunk 1 above)
Comment (test-fullautomation) [reply packagedoc/additional_docs/Description.rst:18]: Hi Son,
I will maintain this.
Thank you,
 Thomas

**Review by test-fullautomation — APPROVED** [review]
Hi Son,
look O.K. for me.
Let us test. I will do further maintenance of the documentation.
Thank you,
 Thomas

**Pull request merged**